strict-version-matcher-plugin/src/main/kotlin/com/google/android/gms/dependencies/Versions.kt

```diff
@@ -0,0 +1,108 @@
+package com.google.android.gms.dependencies
+
+import java.util.regex.Pattern
+
+/**
+ * Allow storing, comparing, and parsing of SemVer version strings.
+ */
+data class SemVerVersionInfo(val major: Int, val minor: Int, val patch: Int) {
+    companion object {
+        /**
+         * Parse a SemVer formatted string to {SemVerVersionInfo}.
+         *
+         * Current implementation is rudimentary, but supports the Google Play
+         * services current use.
+         *
+         * @param[versionString] Three-part Semver string to convert.
+         */
+        fun parseString(versionString:String) : SemVerVersionInfo {
+            val version = versionString.trim();
+            val parts = version.split(".")
+            if (parts.size != 3) {
+                throw IllegalArgumentException("versionString didn't have 3 parts divided by periods.");
+            }
+            val major = Integer.valueOf(parts[0])
+            val minor = Integer.valueOf(parts[1])
+
+            var patchString = parts[2]
+            var dashIndex = patchString.indexOf("-")
+            if (dashIndex != -1) {
```

Comment:
With this handling of possible suffixed versions, doesn't this mean something like 16.0.0 and 16.0.0-SNAPSHOT will be treated the same way? I.e. return the same SemVerVersionInfo object? Should they be?

Maybe SemVerVerisonInfo can have a separate `modifier` member for this purpose? Or is this getting to be YAGNI territory?